pkg/ebpf/telemetry/modifier.go

```diff
@@ -87,11 +107,14 @@ func (t *ErrorsTelemetryModifier) BeforeInit(m *manager.Manager, module names.Mo
 		}
 		log.Tracef("module %s probes %d", module.Name(), opts.MapSpecEditors[helperErrTelemetryMapName].MaxEntries)
 
+		mapNames := getMapNames(m, opts)
+		t.trackedMaps = mapNames
+
 		h := keyHash()
-		for _, ebpfMap := range m.Maps {
+		for _, mapName := range t.trackedMaps {
```

Comment:
i would add a short comment here explaining the bug that was discovered and how this is fixes it

Reply:
`trackedMaps` have been removed https://github.com/DataDog/datadog-agent/pull/32650/commits/776db5cc1ac3f2cc6a657dfbe8c65bd7a3bf95e1